### PR TITLE
Fix a couple of parameters links in the doc

### PR DESCRIPTION
**Pull request by bkhelifi**
A tiny PR....

### CODE REVIEW

File: gammapy/astro/darkmatter/utils.py [hunk 1]
```diff
@@ -12,7 +13,7 @@ class JFactory:
 
     J-Factors are computed for annihilation and D-Factors for decay.
     Set the argument `annihilation` to `False` to compute D-Factors.
-    The assumed dark matter profiles will be centered on the center of the map.
+    The assumed dark matter profiles will be centered in the center of the map.
```
Comment (Astro-Kirsty) [inline gammapy/astro/darkmatter/utils.py:16]: The first one is correct!

File: gammapy/data/event_list.py [hunk 2]
```diff
@@ -528,7 +528,7 @@ def plot_offset2_distribution(
         >>> bins = np.linspace(start=0, stop=0.3 ** 2, num=30) * u.deg ** 2
         >>> events.plot_offset2_distribution(center=center, bins=bins) # doctest: +SKIP
 
-        Note how we passed the ``bins`` option of `matplotlib.pyplot.hist` to control
+        Note how we passed the ``bins`` option of `~matplotlib.pyplot.hist` to control
```
Comment (Astro-Kirsty) [inline gammapy/data/event_list.py:531]: I would leave it as the first option, as it explicitly shows the entire link

File: gammapy/datasets/actors.py [hunk 3]
```diff
@@ -19,15 +19,15 @@ class DatasetsActor(Datasets):
...
+    datasets : `~gammapy.datasets.Datasets`
         Datasets
     """
 
     def __init__(self, datasets=None):
         from ray import get
 
         log.warning(
-            "Gammapy support for parallelisation with ray is still a prototype and is not fully functional."
+            "Gammapy support for parallelization with ray is still a prototype and is not fully functional."
```
Comment (Astro-Kirsty) [inline gammapy/datasets/actors.py:30]:
```suggestion
            "Gammapy support for parallelisation with ray is still a prototype and is not fully functional."
```

File: gammapy/datasets/flux_points.py [hunk 4]
```diff
@@ -71,7 +71,7 @@ class FluxPointsDataset(Dataset):
                     https://iopscience.iop.org/article/10.1088/0004-637X/773/2/168/pdf
 
         Default is `chi2`, in that case upper limits are ignored and the mean of asymetrics error is used.
-        However it is recommended to use `profile` if `stat_scan` is available on flux points.
+        However, it is recommended to use `profile` if `stat_scan` is available on flux points.
```
Comment (Astro-Kirsty) [inline gammapy/datasets/flux_points.py:74]:
```suggestion
        However, it is recommended to use ``profile`` if ``stat_scan`` is available on flux points.
```

File: gammapy/datasets/map.py [hunk 5]
```diff
@@ -1709,15 +1709,15 @@ def info_dict(self, in_safe_data_range=True):
         return info
 
     def to_spectrum_dataset(self, on_region, containment_correction=False, name=None):
-        """Return a ~gammapy.datasets.SpectrumDataset from on_region.
+        """Return a `~gammapy.datasets.SpectrumDataset` from on_region.
```
Comment (Astro-Kirsty) [inline gammapy/datasets/map.py:1712]:
```suggestion
        """Return a `~gammapy.datasets.SpectrumDataset` from ``on_region``.
```

File: gammapy/datasets/map.py [hunk 6]
```diff
@@ -199,7 +199,7 @@ def create_map_dataset_from_observation(
     models : `~gammapy.modeling.Models`, optional
         Models. Default is None.
     dataset_name : str, optional
-        If `models` contains one or multiple `FoVBackgroundModel`
+        If `models` contains one or multiple `~gammapy.modeling.models.FoVBackgroundModel`
         it should match the `dataset_name` of the background model to use.
```
Comment (Astro-Kirsty) [inline gammapy/datasets/map.py:203]:
```suggestion
        If ``models`` contains one or multiple `~gammapy.modeling.models.FoVBackgroundModel`
        it should match the ``dataset_name`` of the background model to use.
```

File: gammapy/datasets/map.py [hunk 7]
```diff
@@ -2867,7 +2867,7 @@ def info_dict(self, in_safe_data_range=True):
         return info
 
     def to_spectrum_dataset(self, on_region, containment_correction=False, name=None):
-        """Return a ~gammapy.datasets.SpectrumDatasetOnOff from on_region.
+        """Return a `~gammapy.datasets.SpectrumDatasetOnOff` from on_region.
```
Comment (Astro-Kirsty) [inline gammapy/datasets/map.py:2870]:
```suggestion
        """Return a `~gammapy.datasets.SpectrumDatasetOnOff` from ``on_region``.
```

File: gammapy/datasets/spectrum.py [hunk 8]
```diff
@@ -334,7 +334,7 @@ def read(cls, filename, format="ogip", checksum=False, **kwargs):
     def write(self, filename, overwrite=False, format="ogip", checksum=False):
         """Write spectrum dataset on off to file.
 
-        Can be serialised either as a `MapDataset` with a `RegionGeom`
+        Can be serialised either as a `~gammapy.datasets.MapDataset` with a `RegionGeom`
```
Comment (Astro-Kirsty) [inline gammapy/datasets/spectrum.py:337]: Should RegionGeom also have a link?

File: gammapy/makers/reduce.py [hunk 9]
```diff
@@ -34,7 +34,7 @@ class DatasetsMaker(Maker, parallel.ParallelMixin):
     cutout_width : tuple of `~astropy.coordinates.Angle`, optional
         Angular sizes of the region in (lon, lat) in that specific order.
         If only one value is passed, a square region is extracted.
-        If None it returns an error, except if the list of makers includes a `SafeMaskMaker`
+        If None it returns an error, except if the list of makers includes a `~gammapy.makers.SafeMaskMaker`
```
Comment (Astro-Kirsty) [inline gammapy/makers/reduce.py:37]:
```suggestion
        If None, an error is returned except if the list of makers includes a `~gammapy.makers.SafeMaskMaker`
```

File: gammapy/modeling/models/spectral.py [hunk 10]
```diff
@@ -659,7 +659,7 @@ def f(x):
     def inverse_all(self, values, energy_min=0.1 * u.TeV, energy_max=100 * u.TeV):
         """Return energies for multiple function values of the spectral model.
 
-        Calls the `scipy.optimize.brentq` numerical root finding method.
+        Calls the `~scipy.optimize.brentq` numerical root finding method.
```
Comment (Astro-Kirsty) [inline gammapy/modeling/models/spectral.py:662]: Again -- leave the full path (i.e. previous implementation)

File: gammapy/modeling/models/spectral.py [hunk 11]
```diff
@@ -1206,9 +1206,9 @@ class PiecewiseNormSpectralModel(SpectralModel):
     ----------
     energy : `~astropy.units.Quantity`
         Array of energies at which the model values are given (nodes).
-    norms : `~numpy.ndarray` or list of `Parameter`
+    norms : `~numpy.ndarray` or list of `~gammapy.modeling.Parameter`
         Array with the initial norms of the model at energies ``energy``.
-        Normalisation parameters are created for each value.
+        Normalization parameters are created for each value.
```
Comment (Astro-Kirsty) [inline gammapy/modeling/models/spectral.py:1211]: en-GB = 's'

**Review by Astro-Kirsty — COMMENTED** [review]
Thanks for the huge effort @bkhelifi. 
Is it necessary to specify the full path if you are in the same file as that function/class?

File: gammapy/modeling/models/core.py [hunk 12]
```diff
@@ -379,7 +379,7 @@ class DatasetModels(collections.abc.Sequence, CovarianceMixin):
 
     Parameters
     ----------
-    models : `SkyModel`, list of `SkyModel` or `Models`
+    models : `~gammapy.modelling.SkyModels`, list of `SkyModel` or `Models`
```
Comment (registerrier) [inline gammapy/modeling/models/core.py:382]:
```suggestion
    models : `SkyModel`, list of `SkyModel` or `Models`
```

File: gammapy/modeling/models/spatial.py [hunk 13]
```diff
@@ -1738,7 +1738,7 @@ def evaluate(self, lon, lat, energy=None, **norms):
         lon = Angle(lon).wrap_at(0 * u.deg)
         lon = Angle(lon).wrap_at(self._wrap_angle)
         # by default rely on CloughTocher2DInterpolator
-        # (Piecewise cubic, C1 smooth, curvature-minimizing interpolant)
+        # (Piecewise cubic, C1 smooth, curvature-minimizing interpolation)
```
Comment (registerrier) [inline gammapy/modeling/models/spatial.py:1741]:
```suggestion
        # (Piecewise cubic, C1 smooth, curvature-minimizing interpolant)
```

File: gammapy/modeling/models/temporal.py [hunk 14]
```diff
@@ -934,14 +935,14 @@ class TemplatePhaseCurveTemporalModel(TemporalModel):
...
-    The phasecurve is given as a table with columns ``phase`` and ``norm``.
+    The phase curve is given as a table with columns ``phase`` and ``norm``.
 
     The ``norm`` is supposed to be a unit-less multiplicative factor in the model,
     to be multiplied with a spectral model.
 
     The model does linear interpolation for times between the given ``(phase, norm)`` values.
 
-    The implementation currently uses `scipy.interpolate. InterpolatedUnivariateSpline`,
+    The implementation currently uses `scipy.interpolate.InterpolatedUnivariateSpline`,
```
Comment (registerrier) [inline gammapy/modeling/models/temporal.py:945]:
```suggestion
    The implementation currently uses `~scipy.interpolate.InterpolatedUnivariateSpline`,
```

**Review by registerrier — COMMENTED** [review]
Thanks @bkhelifi . In practice keeping `~ClassName` works well in the same subpackage. It can have the advantage of working well if we rename the subpackage. Technically, many links you correct were already working.

Now, this is not uniform in the documentation and we have not made a specific choice as how to do this in general. Are there opinions @AtreyeeS @QRemy @Astro-Kirsty ?

File: gammapy/data/metadata.py [hunk 15]
```diff
@@ -122,7 +122,7 @@ class GTIMetaData(MetaData):
 
     Parameters
     ----------
-    reference_time : Time, str
+    reference_time : str
```
[the remark this answers is not in the document]
Comment (bkhelifi) [reply gammapy/data/metadata.py:125]: It was like this, but @Astro-Kirsty prefered like this...

File: gammapy/modeling/models/temporal.py (hunk 14 above)
Comment (bkhelifi) [reply gammapy/modeling/models/temporal.py:945]: It was like this, but @Astro-Kirsty prefered like this...

File: gammapy/modeling/models/core.py (hunk 12 above)
Comment (bkhelifi) [reply gammapy/modeling/models/core.py:382]: Why not putting the link towards the documentation? while in general all parameters are links...

**Comment by Astro-Kirsty** [comment]
> Thanks @bkhelifi . In practice keeping `~ClassName` works well in the same subpackage. It can have the advantage of working well if we rename the subpackage. Technically, many links you correct were already working.

Yes, this is what I was also mentioning in my comment. I think we should leave it as it was with `~ClassName` without the entire path.

**Review by morcuended — COMMENTED** [review]
Thanks for the huge effort @bkhelifi. I left a couple of minor comments.

Regarding whether to use `~ClassName` or the full path, I would also use simply `~ClassName` for better readability. I would keep the full path when the context is not clear and you need to specify where the method is coming from.

**Comment by registerrier** [comment]
Regarding the way we write argument types. Maybe it is useful to check the Sphinx documentation. See:
- [target specification](https://www.sphinx-doc.org/en/master/usage/domains/python.html#target-specification)
- [target resolution](https://www.sphinx-doc.org/en/master/usage/domains/python.html#target-resolution)

From the former the `~` prefix, means that what is shown in the doc is only the last part of the path e.g. `~gammapy.maps.MapAxis` will render `MapAxis` with a link. So `~MapAxis` is probably not needed.

From the latter,  it is written that: `Names in these roles are searched first without any further qualification, then with the current module name prepended, then with the current module and class name (if any) prepended.`
So we don't have to put the full path when the documented object is in the current module. 

External links to other packages docs is done through the intersphinx mappings defined in conf.py. 

The LSST dev doc is also useful [here](https://developer.lsst.io/restructuredtext/style.html#links-to-python-objects).  Note that it suggests to always provide the full namespace [here](https://developer.lsst.io/python/numpydoc.html#concrete-types)

We have to decide on a systematic approach, document it in the dev doc and implement it consistently in docstrings (code reviews in particular have to be clear on this).

My opinion on this is that we should stick to what we have implemented most of time so far: full namespace links with `~` prefix. 

Opinions?

File: gammapy/modeling/models/core.py (hunk 12 above)
Comment (registerrier) [reply gammapy/modeling/models/core.py:382]: Right but here you introducing an incorrect namespace. In current doc all links are OK. See https://docs.gammapy.org/dev/api/gammapy.modeling.models.DatasetModels.html

File: gammapy/modeling/models/temporal.py (hunk 14 above)
Comment (registerrier) [reply gammapy/modeling/models/temporal.py:945]: OK. The change I propose would remove the full path in the doc, while here I think we need it. So don't apply.

**Comment by morcuended** [comment]
> My opinion on this is that we should stick to what we have implemented most of time so far: full namespace links with ~ prefix, when linking to classes in arguments description.

> For methods or attributes inside a class, this is less clear. See for instance the proposed change:
Normalization used to compute the residuals, see `FluxPointsDataset.residuals`. to Normalization used to compute the residuals, see `~gammapy.datasets.FluxPointsDataset.residuals`.
The issue here is that only residuals will show in the final docs, while it would be useful to have the explicit class name where this belongs, i.e. FluxPointsDataset.residuals as is the case in the [current doc version](https://docs.gammapy.org/dev/api/gammapy.datasets.FluxPointsDataset.html#gammapy.datasets.FluxPointsDataset.plot_residuals).

Thanks for the references @registerrier. I agree with your proposal. We should include these guidelines in the dev doc as you propose.

**Review by registerrier — COMMENTED** [review]
Thanks @bkhelifi .  So finally, you implemented the opposite of what I suggested to limit changes in this [comment](https://github.com/gammapy/gammapy/pull/5490#issuecomment-2414831951). 
So we really have to define a clear guideline. 

Reviewing 128 modified files is practically impossible unfortunately.

**Pull request closed without merging**